### PR TITLE
[SYCL] [USM] Implement prefetch.

### PR DESCRIPTION
**Pull request by jcranmer-intel**
Signed-off-by: Joshua Cranmer <joshua.cranmer@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
@jcranmer-intel, are you able to address the comments from @romanovvlad?
Do you need any help?

**Comment by jcranmer-intel** [comment]
Oops, completely forgot I still had this outstanding. I'll get a new patch up for you.

**Comment by jbrodman** [comment]
Could you also enable prefetch for the HOST device? See https://github.com/intel/llvm/pull/579.

I imagine it can just be a no-op on HOST for now.

**Comment by jbrodman** [comment]
(and support ordered_queue too?) (sorry!)

File: sycl/source/detail/memory_manager.cpp [hunk 1]
```diff
@@ -500,6 +500,21 @@ void MemoryManager::fill_usm(void *Mem, QueueImplPtr Queue, size_t Length,
   }
 }
 
+void MemoryManager::prefetch_usm(void *Mem, QueueImplPtr Queue, size_t Length,
+                                std::vector<RT::PiEvent> DepEvents,
+                                RT::PiEvent &OutEvent) {
+  sycl::context Context = Queue->get_context();
+
+  if (Context.is_host()) {
+    // TODO: Potentially implement prefetch on the host.
```
Comment (romanovvlad) [inline sycl/source/detail/memory_manager.cpp]: We have some implementation of prefetch for host already. ./source/detail/platform_util.cpp:123.
Haven't check if it's suitable here.

**Pull request merged**